### PR TITLE
POM editing: add ability to some mojos to "edit" the POM

### PR DESCRIPTION
**Pull request by cstamas**
Introduce a Sink<Artifact> that collects transformations based on received artifacts and applies it onto POM.

Currently three usecases:
* `gav-list-available-plugins` is now able to generate POM with discovered plugins as `pluginManagement`.
* `versions` is now able (in somewhat limited way) to apply discovered updates onto POM.
* `plugin-versions` same

### CODE REVIEW

File: shared/src/main/java/eu/maveniverse/maven/toolbox/shared/internal/PomTransformerSink.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                        StandardCopyOption.REPLACE_EXISTING);
+            }
+        } catch (IOException ex) {
+            // ignore
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        new PomTransformer(pom, StandardCharsets.UTF_8, PomTransformer.SimpleElementWhitespace.EMPTY)
```
Comment (ppalaga) [inline shared/src/main/java/eu/maveniverse/maven/toolbox/shared/internal/PomTransformerSink.java:155]: `SimpleElementWhitespace.AUTODETECT_PREFER_EMPTY` might be more respectful to folks having other preferences.

**Comment by cstamas** [comment]
Example (as in IT) of intput and output:
https://gist.github.com/cstamas/bb5065d07f5be66a3daf1d361e79ca5a

`versions -DapplyToPom`

**Comment by cstamas** [comment]
Created issue, very similar to what code in this hacked PR does https://github.com/l2x6/pom-tuner/issues/159

**Comment by cstamas** [comment]
Example of current master Maven POM: https://gist.github.com/cstamas/8cbf7f7f4d73e1c53647d9bcedf99c45

**Pull request merged**